### PR TITLE
Rewriting existing tests using Node.js Test Runner

### PR DESCRIPTION
**Pull request by almilosz**
### Details:
 - Rewriting existing tests using Node.js Test Runner
 - Removing 3rd party module JEST and its dependencies

### Tickets:
 - [117405]

### CODE REVIEW

File: src/bindings/js/node/tests/basic.test.js [hunk 1]
```diff
@@ -1,4 +1,6 @@
-const ov = require('../build/Release/ov_node_addon.node');
+let ov = require('../build/Release/ov_node_addon.node');
+const assert = require('assert');
+const {describe, it} = require('node:test');
```
Comment (vishniakov-nikolai) [inline src/bindings/js/node/tests/basic.test.js:3]:
```suggestion
const { describe, it } = require('node:test');
```

File: src/bindings/js/node/tests/basic.test.js [hunk 2]
```diff
@@ -13,119 +15,118 @@ function getModelPath(isFP16=false) {
...
   );
 
-  test.each([
-    tensor,
-    tensor_data,
-  ])('Different tensor-like values', (val) => {
+  const tensorLike = [[tensor],
+    [tensorData]];
+
+  tensorLike.forEach(([tl]) => {
```
Comment (vishniakov-nikolai) [inline src/bindings/js/node/tests/basic.test.js:111]:
```suggestion
  tensorLike.forEach(tl => {
```

File: src/bindings/js/node/tests/tensor.test.js [hunk 3]
```diff
@@ -1,113 +1,113 @@
 var ov = require('../build/Release/ov_node_addon.node');
+const assert = require("assert");
+const {test, describe, it} = require("node:test");
```
Comment (vishniakov-nikolai) [inline src/bindings/js/node/tests/tensor.test.js:3]: Looks like eslint doesn't check formatting 
```suggestion
const assert = require('assert');
const { test, describe, it } = require('node:test');
```

File: src/bindings/js/node/tests/tensor.test.js [hunk 4]
```diff
@@ -1,113 +1,113 @@
...
-    it.each([
-        [ov.element.f32],
-        [ov.element.i32],
-        [ov.element.u32],
-    ])('compares ov.element %p', (ov_elem) => {
-        var tensor = new ov.Tensor(ov_elem, Int32Array.from([1, 224, 224, 3]), new Float32Array(150528));
-        expect(tensor.getPrecision()).toEqual(ov_elem);
+  });
+
+  ets.forEach(([elemType, ..._]) => {
```
Comment (vishniakov-nikolai) [inline src/bindings/js/node/tests/tensor.test.js:107]: Just skip the rest of parameters
```suggestion
  ets.forEach(([elemType]) => {
```

**Review by vishniakov-nikolai — APPROVED** [review]
Good job!

File: src/bindings/js/node/tests/tensor.test.js (hunk 4 above)
Comment (almilosz) [reply src/bindings/js/node/tests/tensor.test.js:107]: Array destructuring in JS never fails to surprise me

**Pull request merged**